lstchain/calib/camera/r0.py

```diff
@@ -253,21 +253,20 @@ def interpolate_pseudo_pulses(waveform, expected_pixel_id, fc, fc_old, n_modules
                         abspos = int(1024 - roisize - 2 - fc_old[nr_module, gain, pix] + k * 1024 + size4drs)
                         spike_A_position = int((abspos - fc[nr_module, gain, pix] + size4drs) % size4drs)
                         if (spike_A_position > 2 and spike_A_position < 38):
-                            pixel = expected_pixel_id[nr_module*7 + pix]
-                            interpolate_spike_A(waveform, gain, spike_A_position, pixel)
+                            # The correction is only needed for even last capacitor (lc) in the
+                            # first half of the DRS4 ring
+                            if (fc_old[nr_module, gain, pix] + 39) % 2 == 0 and (fc_old[nr_module, gain, pix] + 39) % 1024 <= 510:
+                                pixel = expected_pixel_id[nr_module*7 + pix]
+                                interpolate_spike_A(waveform, gain, spike_A_position, pixel)
                         # looking for spike A second case
                         abspos = int(roisize - 2 + fc_old[nr_module, gain, pix] + k * 1024)
                         spike_A_position = int((abspos - fc[nr_module, gain, pix] + size4drs) % size4drs)
                         if (spike_A_position > 2 and spike_A_position < 38):
-                            pixel = expected_pixel_id[nr_module*7 + pix]
-                            interpolate_spike_A(waveform, gain, spike_A_position, pixel)
-
-                    # looking for spike B
-                    spike_b_position = int(
-                        (fc_old[nr_module, gain, pix] - 1 - fc[nr_module, gain, pix] + 2 * size4drs) % size4drs)
-                    if spike_b_position < roisize - 1:
-                        pixel = expected_pixel_id[nr_module*7 + pix]
-                        interpolate_spike_B(waveform, gain, spike_b_position, pixel)
+                            # The correction is only needed for even last capacitor (lc) in the
+                            # first half of the DRS4 ring
+                            if (fc_old[nr_module, gain, pix] + 39) % 2 == 0 and (fc_old[nr_module, gain, pix] + 39) % 1024 <= 510:
```

Comment:
numbers like 38, 39 could be `len_drs4_window` (or something like that)

Reply:
done